lib/generate-doc.js

```diff
@@ -15,6 +15,9 @@ module.exports = function generateDocument (baseDocument, router) {
   // Iterate the middleware stack and add any paths and schemas, etc
   router && router.stack.forEach((_layer) => {
     iterateStack('', null, _layer, (path, routeLayer, layer) => {
+      if (basePath) {
+        path = path.replace(basePath, '')
```

Comment:
I am pretty sure this needs strip it only from the front. In this case if we had a base path of `/api` and a path of `/bar/foo/api` I believe we would end up with `/bar/foo` instead of `/bar/foo/api`.

Reply:
Yes! Should be handled in 639d4524a22a670c8d467aa06ade54ac0976fa56